### ISSUE TITLE
Temperature sampling should affect the probability distribution returned as well

### ISSUE THREAD

**Issue by prajwal1210** [axonn-ai/yalis#32]
I noticed that vLLM applies temperature = 0 by actually making the logits to a one-hot vector and the model output actually returns this updated tensor: https://github.com/vllm-project/vllm/blob/d84cef76eb9e16190cfdd97ae24511c8c819f179/vllm/v1/sample/sampler.py#L62

Not sure what the "true" definition of temperature should do, but maybe we ought to do the same.

**Comment by prajwal1210** [comment]
This is only true for SpecDec and we have implemented a separate function to compute probs from logits with temperature scaling applied.
